mixnet/client/src/sender.rs

```diff
@@ -109,12 +129,15 @@ impl<R: Rng> Sender<R> {
 
         let mu: std::sync::Arc<tokio::sync::Mutex<tokio::net::TcpStream>> =
             pool.get_or_init(&addr).await?;
+        let arc_socket = mu.clone();
         let mut socket = mu.lock().await;
-        let body = Body::new_sphinx(packet);
-        body.write(&mut *socket).await?;
-
-        tracing::debug!("Sent a Sphinx packet successuflly to the node: {addr:?}");
+        let body = Body::SphinxPacket(packet);
 
+        if let Err(e) = body.write(&mut *socket).await {
+            tracing::error!("Failed to send packet to {addr} with error: {e}. Retrying...");
+            return mixnet_protocol::retry_backoff(addr, max_retries, retry_delay, body, arc_socket)
+            .await;
```

Comment:
Do we probably have to drop `socket` first before calling `retry_backoff`? as you did at the other part.

Reply:
Ah yes, thank you!